### PR TITLE
Redirect generators to a temporary directory in test mode

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by poettering** [comment]
lgtm

File: src/shared/path-lookup.h [hunk 1]
```diff
@@ -60,6 +61,9 @@ struct LookupPaths {
 
         /* The root directory prepended to all items above, or NULL */
         char *root_dir;
+
+        /* A temporary directory when running in test mode, to be nuked */
+        char *temporary_dir;
```
Comment (evverx) [inline src/shared/path-lookup.h:66]: This one doesn't seem to be freed. I've started getting the following error recently:
```sh
==17045==
==17045== 30 bytes in 1 blocks are definitely lost in loss record 1 of 5
==17045==    at 0x4C2DB9D: malloc (vg_replace_malloc.c:299)
==17045==    by 0x68E0CC9: strdup (in /usr/lib64/libc-2.24.so)
==17045==    by 0x23D7AC: mkdtemp_malloc (fileio.c:1517)
==17045==    by 0x219126: lookup_paths_init (path-lookup.c:491)
==17045==    by 0x14F0F8: manager_startup (manager.c:1315)
==17045==    by 0x132B24: main (test-sched-prio.c:48)
==17045==
```

Comment (keszybz) [reply src/shared/path-lookup.h:66]: There's a fix queued up in #6847.

Comment (evverx) [reply src/shared/path-lookup.h:66]: I can't keep up with you :-) I'm sorry for not noticing the fix.

**Pull request merged**